### PR TITLE
[v0.16.x] Backport more PRs to 0.16

### PR DESCRIPTION
**Pull request by PastaPastaPasta**
This backports: 

#3625 
#3629 
#3656 
#3662 
#3666 
#3669 
#3672 
#3685 
#3687 
as well as a partial backport of #3636

### CODE REVIEW

File: src/llmq/quorums_signing_shares.cpp [hunk 1]
```diff
@@ -677,7 +677,7 @@ bool CSigSharesManager::ProcessPendingSigShares(CConnman& connman)
             if (!pubKeyShare.IsValid()) {
                 // this should really not happen (we already ensured we have the quorum vvec,
                 // so we should also be able to create all pubkey shares)
-                LogPrintf("CSigSharesManager::%s -- pubKeyShare is invalid, which should not be possible here");
+                LogPrintf("CSigSharesManager::%s -- pubKeyShare is invalid, which should not be possible here", __func__);
```
Comment (UdjinM6) [inline src/llmq/quorums_signing_shares.cpp:680]: It looks like a prior fix is missing here (`\n` at the end). Might also want to (partially) backport https://github.com/dashpay/dash/pull/3520/commits/ec7c7dfc8d18c22754a432e781192d0b31eadadd and https://github.com/dashpay/dash/pull/3520/commits/2beab95acb7e125fb9f122b11f25434af203c471 from #3520 to fix similar issues in other places.

**Review by UdjinM6 — COMMENTED** [review]
Might be a good idea to backport some other related fixes, see below. Looks good otherwise.

File: src/llmq/quorums_signing_shares.cpp (hunk 1 above)
Comment (PastaPastaPasta) [reply src/llmq/quorums_signing_shares.cpp]: sure. Cherry-picked both

**Review by UdjinM6 — APPROVED** [review]
LGTM, utACK

**Review by xdustinface — APPROVED** [review]
Looks good, utACK

**Comment by PastaPastaPasta** [comment]
Tag removed from all backported PRs

**Pull request merged**